### ISSUE TITLE
Add a reduce method

### ISSUE THREAD

**Issue by jorainer** [rformassspectrometry/Spectra#282]
Add a method `reduce` that groups all peaks within each spectrum based on user provided `ppm` and `tolerance` values and keep for each group only the peak with the highest intensity. Could be alternatively called `prune`, but I think `reduce` captures the meaning of the function better, i.e. it's purpose is to *clean*/*simplify* each spectrum removing peaks with very similar m/z.

**Comment by lgatto** [comment]
I would suggest to call it `reduceSpectra()` - reduce is too generic.

**Comment by lgatto** [comment]
I don't think we should should aim for generality in `Spectra` and would prefer `reduceSpectra` along the lines of longer, more informative names such as `filterPrecursorMz()`, ... 

I think we can have shorter names in lower-level function/packages such as `MsCoreUtils` and `MetaboCoreUtils`, as they work on more general/basic data types, and are aimed for developers, who can be expected to know the nature of the data/objects they manipulate.

**Comment by AharoniLab** [comment]
'reduceSpectra' sounds just sufficiently focused in my view.
